build_tools/benchmarks/run_benchmarks_on_linux.py

```diff
@@ -38,48 +43,69 @@ def run_benchmark_case(self, benchmark_case: BenchmarkCase,
                          benchmark_results_filename: Optional[pathlib.Path],
                          capture_filename: Optional[pathlib.Path]) -> None:
 
-    # TODO(pzread): Taskset should be derived from CPU topology.
-    # Only use the low 8 cores.
-    taskset = "0xFF"
-
-    # TODO(#11076): Support run_config.
-    if benchmark_case.benchmark_case_dir is None:
-      raise ValueError("benchmark_case_dir can't be None.")
-
-    run_flags = self.__parse_flagfile(benchmark_case.benchmark_case_dir)
-    # Replace the CUDA device flag with the specified GPU.
-    if benchmark_case.driver_info.driver_name == "cuda":
-      run_flags = list(
-          filter(lambda flag: not flag.startswith("--device"), run_flags))
-      run_flags.append(f"--device=cuda://{self.gpu_id}")
-
     if benchmark_results_filename:
       self.__run_benchmark(benchmark_case=benchmark_case,
-                           results_filename=benchmark_results_filename,
-                           taskset=taskset,
-                           run_flags=run_flags)
+                           results_filename=benchmark_results_filename)
 
     if capture_filename:
       self.__run_capture(benchmark_case=benchmark_case,
-                         capture_filename=capture_filename,
-                         taskset=taskset,
-                         run_flags=run_flags)
+                         capture_filename=capture_filename)
 
   def __parse_flagfile(self, case_dir: pathlib.Path) -> List[str]:
     return [
         line.strip()
         for line in (case_dir / MODEL_FLAGFILE_NAME).read_text().splitlines()
     ]
 
+  def __build_tool_cmds(self, benchmark_case: BenchmarkCase,
+                        tool_path: pathlib.Path) -> List[Any]:
+    run_config = benchmark_case.run_config
+    if run_config is None:
+      # TODO(#11076): Remove legacy path.
+      if benchmark_case.benchmark_case_dir is None:
+        raise ValueError(
+            "benchmark_case_dir can't be None if run_config is None.")
+
+      # TODO(pzread): Taskset should be derived from CPU topology.
+      # Only use the low 8 cores.
+      cmds = ["taskset", "0xFF", tool_path]
+
+      run_flags = self.__parse_flagfile(benchmark_case.benchmark_case_dir)
+      # Replace the CUDA device flag with the specified GPU.
+      if benchmark_case.driver_info.driver_name == "cuda":
+        run_flags = list(
+            filter(lambda flag: not flag.startswith("--device"), run_flags))
+        run_flags.append(f"--device=cuda://{self.gpu_id}")
+
+      return cmds + run_flags
+
+    cmds: List[Any] = run_module_utils.build_linux_wrapper_cmds_for_device_spec(
+        run_config.target_device_spec)
+    cmds += [tool_path]
```

Comment:
append?

```suggestion
    cmds.append(tool_path)
```

(same below)